ports/nrf/modules/board/led.c

```diff
@@ -33,48 +33,140 @@
 
 #if MICROPY_HW_HAS_LED
 
-#define LED_OFF(pin) {(MICROPY_HW_LED_PULLUP) ? nrf_gpio_pin_set(pin) : nrf_gpio_pin_clear(pin); }
-#define LED_ON(pin) {(MICROPY_HW_LED_PULLUP) ? nrf_gpio_pin_clear(pin) : nrf_gpio_pin_set(pin); }
-
 typedef struct _board_led_obj_t {
     mp_obj_base_t base;
     mp_uint_t led_id;
     mp_uint_t hw_pin;
     uint8_t hw_pin_port;
+    bool act_level;
 } board_led_obj_t;
 
-STATIC const board_led_obj_t board_led_obj[] = {
+static inline void LED_OFF(board_led_obj_t * const led_obj) { 
```

Comment:
As this function replaces the Macro with the same name, lower case would look better. The same applies for LED_ON.

Reply:
@glennrub Yes it is to replace the MACRO.  I kept the Upper case just for compatibility with the old boards.  It would be nicer in lowercase of course.

Reply:
Ah, i see. The good thing is that this is an macro internal to the .c file, so there are no other users than the module itself. Safe to lower-case. :)